tools/cloud_functions/gcs_event_based_ingest/backfill.py

```diff
@@ -0,0 +1,175 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Command Line utility for backfilling gcs_ocn_bq_ingest cloud function
+"""
+import concurrent.futures
+import logging
+import os
+import pprint
+import sys
+from argparse import ArgumentParser, Namespace
+from typing import Dict, Iterator, List
+
+from google.api_core.client_info import ClientInfo
+from google.cloud import storage
+
+import gcs_ocn_bq_ingest  # pylint: disable=import-error
+
+CLIENT_INFO = ClientInfo(user_agent="google-pso-tool/bq-severless-loader")
+
+os.environ["FUNCTION_NAME"] = "backfill-cli"
+
+
+def find_blobs_with_suffix(
+    gcs_cli: storage.Client,
+    prefix: str,
+    suffix: str = "_SUCCESS",
+) -> Iterator[storage.Blob]:
+    """
+    Find GCS blobs with a given suffix.
+
+    :param gcs_cli:  storage.Client
+    :param prefix: A GCS prefix to search i.e. gs://bucket/prefix/to/search
+    :param suffix: A suffix in blob name to match
+    :return:  Iterable of blobs matching the suffix.
+    """
+    bucket_name, prefix = gcs_ocn_bq_ingest.main.parse_gcs_url(
+        prefix)    # noqa
+    bucket: storage.Bucket = gcs_cli.lookup_bucket(bucket_name)
+    # filter passes on scalability / laziness advantages of iterator.
+    return filter(lambda blob: blob.name.endswith(suffix),
+                  bucket.list_blobs(prefix=prefix))
+
+
+def main(args: Namespace):
+    """main entry point for backfill CLI."""
+    gcs_cli: storage.Client = storage.Client(client_info=CLIENT_INFO)
+    ps_cli = None
```

Comment:
Rename to a more descriptive variable name like "pubsub_client". 
cli is more commonly understood as "command line interface"